examples/api/lib/painting/star_border/star_border.0.dart

```diff
@@ -0,0 +1,445 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// An example showing usage of [StarBorder].
+
+import 'package:flutter/material.dart';
+
+const int _kParameterPrecision = 2;
+
+void main() => runApp(const StarBorderApp());
+
+class StarBorderApp extends StatelessWidget {
+  const StarBorderApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('StarBorder Example'),
+          backgroundColor: const Color(0xff323232),
+        ),
+        body: const StarBorderExample(),
+      ),
+    );
+  }
+}
+
+class StarBorderExample extends StatefulWidget {
+  const StarBorderExample({super.key});
+
+  @override
+  State<StarBorderExample> createState() => _StarBorderExampleState();
+}
+
+class _StarBorderExampleState extends State<StarBorderExample> {
+  final OptionModel _model = OptionModel();
+  final TextEditingController textController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _model.addListener(_modelChanged);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _model.removeListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return DefaultTextStyle(
+      style: const TextStyle(
+        color: Colors.black,
+        fontSize: 14.0,
+        fontFamily: 'Roboto',
+        fontStyle: FontStyle.normal,
+      ),
+      child: SliderTheme(
+        data: SliderTheme.of(context).copyWith(),
+        child: ListView(
+          children: <Widget>[
+            Container(
+              color: Colors.grey.shade200,
+              child: Options(_model),
+            ),
+            Padding(
+              padding: const EdgeInsets.all(18.0),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: <Widget>[
+                  Expanded(
+                    child: Container(
```

Comment:
Could factor out a widget here and below, like:
```dart
class StarBorderView extends StatelessWidget {
  const StarBorderView(this.star, this.title);

  final StarBorder star;
  final String title;

  @override
  Widget build(BuildContext) {
    return Container(
      key: UniqueKey(),
      alignment: Alignment.center,
      padding: const EdgeInsets.all(20),
      width: 150,
      height: 100,
      decoration: ShapeDecoration(
        color: Colors.blue.shade100,
        shape: star,
      ),
      child: const Text(title),
    );
  }
}
```

There are other places where you could do this kind of thing. Not a big deal.